src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1043,13 +1043,14 @@ private boolean shouldDumpItem(
             returnStack = InventoryUtils.addItemStackToProviderWithResult(buildingWorker.getTileEntity(), tempStack);
         }
 
-        new InvWrapper(worker.getInventoryCitizen()).extractItem(slot, stack.stackSize - amountToKeep, false);
-
         if (returnStack == null)
         {
+            new InvWrapper(worker.getInventoryCitizen()).extractItem(slot, stack.stackSize - amountToKeep, false);
```

Comment:
It would have looked better if they made it something like:
`InvWrapper.create(IInventory inventory)` instead of having to use the constructor. But hey this works too.